pyfolio/sanitize.py

```diff
@@ -0,0 +1,266 @@
+#
+# Copyright 2018 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from warnings import warn
+import pandas as pd
+
+
+def sanitize(returns=None,
+             positions=None,
+             txns=None):
+    """
+    Sanitize inputs to pyfolio.
+
+    Returns
+    -------
+    returns, positions, txns : pd.Series or pd.DataFrame
+        Sanitized inputs to pyfolio create_x_tear_sheet functions.
+        See pyfolio.create_full_tear_sheet for more details.
+
+    Usage
+    -----
+    To sanitize all inputs:
+        rets, pos, txns = sanitize_inputs(rets, pos, txns)
+    To sanitize only e.g. positions:
+        _, pos, _ = sanitize_inputs(positions=pos)
+    """
+    if returns is not None:
+        returns = sanitize_returns(returns)
+
+    if positions is not None:
+        positions = sanitize_positions(positions)
+
+    if txns is not None:
+        txns = sanitize_txns(txns)
+
+    return returns, positions, txns
+
+
+def sanitize_returns(returns):
+    """
+    Sanitize returns.
+
+    Returns
+    -------
+    returns : pd.Series
+        Daily returns of the strategy, noncumulative.
+         - Time series with decimal returns.
+         - Example:
+            2015-07-16    -0.012143
+            2015-07-17    0.045350
+            2015-07-20    0.030957
+            2015-07-21    0.004902
+
+    Notes
+    -----
+    - Must be a pd.Series
+        - If it is a 1D dataframe, squeeze and warn.
+        - Otherwise, raise a ValueError
+    - There must not be any NaNs
+        - If there are, raise a ValueError
+    - Index must be timezone-localized datetimes
+        - If not, localize to UTC and warn.
+    - Returns must be floats
+        - If not, coerce values to float and warn
+        - Failing that, raise a ValueError
+    """
+    try:
+        assert isinstance(returns, pd.Series)
```

Comment:
@eigenfoo it's almost never a good idea to use assertions for control flow. Assertions are turned off if you run python with the `-O` flag.  I think for all of these checks you just want plain old if-statements.

Reply:
Ah I see! What's does proper usage of assertions look like, then? Google says for debugging purposes, and for checks that aren't performance-critical (i.e. check if things are _not_ the way you want them, rather than checking if they _are_). Does that sound right?